optuna/storages/_heartbeat.py

```diff
@@ -100,3 +101,14 @@ def fail_stale_trials(study: "optuna.Study") -> None:
         for trial_id in failed_trial_ids:
             failed_trial = copy.deepcopy(storage.get_trial(trial_id))
             failed_trial_callback(study, failed_trial)
+
+
+def is_heartbeat_enabled(storage: BaseStorage) -> bool:
+    """Check whether the storage enables the heartbeat.
+
+    Returns:
+        :obj:`True` if the storage also inherits :class:`~optuna.storages._heartbeat.BaseHeartbeat`
+        and the return value of :meth:`~optuna.storages.BaseStorage.get_heartbeat_interval` is an
+        integer, otherwise :obj:`False`.
+    """
+    return isinstance(storage, BaseHeartbeat) and storage.get_heartbeat_interval() is not None
```

Comment:
```suggestion
    return isinstance(storage, BaseHeartbeat) and storage.get_heartbeat_interval() is not None

BaseStorage.is_heartbeat_enabled = is_heartbeat_enabled

```

From users' perspective, I think `storage.is_heartbeat_enabled()` is more intuitive than `is_heartbeat_enabled(storage)`. So, what about patching `BaseStorage` class? This will remove the cyclic dependency while still keeping the syntax `storage.is_heartbeat_enabled()`.

Of course this is subject to discussion because it is not standard that a bound function is not defined in the class.

Reply:
I understand that there are arguments both ways, so I don't retract my approval.

Reply:
Hmm, even if providing `BaseStorage.is_heartbeat_enabled()` is more intuitive, I think we need to discuss some design trade-off.
- Since `BaseStorage` currently is not open for library users, we can prioritize simplicity over intuitiveness.
- Patching `BaseStorage` in `_heartbeat.py` might be somewhat confusing in terms of the library structure.

Can we just leave those discussion out-of-scope of this PR so far?

Reply:
I think that's a valid argument. I'll merge this PR.